pkg/lockfile/parse-pnpm-lock.go

```diff
@@ -64,6 +64,13 @@ func extractPnpmPackageNameAndVersion(dependencyPath string) (string, string) {
 		version = parts[0]
 	}
 
+	// handle pnpm v6.0+ package specifier that uses name@version pattern
+	atName := strings.Split(name, "@")
```

Comment:
Does this handle packages with "@" in the name?

Reply:
That should be handled due to line 53, but we should double check with a v6 specific test

Reply:
Looks like it was a good idea to add the additional test case as it did have a bug when the package name contained an "@".

I have added this test case and updated my implementation to match the "name@version" pattern explicitly, but allow name to contain zero or more "@" letters in it.

Many of the fixtures used in `parse-pnpm-lock_test.go` tests are using lock file version `5.3`. To ensure there are no further bugs with the implementation, should I create a duplicate of each using current PNPM version? (or would this be going overboard as it would be adding an additional 12 or so test cases)

Reply:
What I usually do is put a handful of the different types in the single test, focusing on the ones that are most likely to be impacted (in particular, we probably don't need to have multiple tests for different number of packages)

Reply:
I think that the two test cases included in this PR are sufficient for now (please let me know if there are other test scenarios you would like me to add). Additionally, I have built and tested the osv-scanner with a OSS project I am working on that uses the v6.0 lock file and lots of dependencies and everything looks ok.